quarkc/lib/primitives.q

```diff
@@ -396,6 +396,23 @@ namespace quark {
         macro JSONObject __to_JSONObject() self.toJSON();
     }
 
+    // TODO:
+    //  - maybe change all logic to be if is-a-quark-object then doit, else go native
+    //  - move macros into native portion of library (avoid double evals)
+    //  - pick better names
+    //  - later on could make blocks work with a little compiler magic to prepend & if the last arg is a Callable
+
+    interface UnaryCallable {
```

Comment:
Minimal documentation would be nice (from both Quark developer perspective and native language perspective).